ql/instruments/creditdefaultswap.cpp

```diff
@@ -263,41 +329,67 @@ namespace QuantLib {
                 FlatHazardRate(0, WeekendsOnly(),
                                Handle<Quote>(flatRate), dayCounter)));
 
-        MidPointCdsEngine engine(probability, recoveryRate, discountCurve);
-        setupArguments(engine.getArguments());
+        boost::shared_ptr<PricingEngine> engine;
+        if(useIsdaEngine) {
+            engine.reset(new IsdaCdsEngine(
+                             probability, recoveryRate, discountCurve,
+                             boost::none,
+                             IsdaCdsEngine::Taylor,
+                             IsdaCdsEngine::HalfDayBias,
+                             IsdaCdsEngine::Piecewise));
+        } else {
+            engine.reset(new MidPointCdsEngine(
+                        probability, recoveryRate, discountCurve));
+        }
+        setupArguments(engine->getArguments());
         const CreditDefaultSwap::results* results =
             dynamic_cast<const CreditDefaultSwap::results*>(
-                                                       engine.getResults());
+                engine->getResults());
 
-        ObjectiveFunction f(targetNPV, *flatRate, engine, results);
-        Rate guess = 0.001;
-        Rate step = guess*0.1;
-
-        return Brent().solve(f, accuracy, guess, step);
+        ObjectiveFunction f(targetNPV, *flatRate, *engine, results);
+        //very close guess if targetNPV = 0.
+        Rate guess = std::log(runningSpread_ / (1 - recoveryRate) * 365./360.);
+        Real step = 0.2; //multiplicative step, equivalent to exp(0.2) factor
```

Comment:
Did you find the search to work better in log space? Just curious...

Reply:
I've reverted it. The accuracy parameter can be larger in log space, but  I haven't checked that it actually reduces the number of step.